types/gauge/index.d.ts

```diff
@@ -0,0 +1,129 @@
+        getTheme(name: string): Theme;
+        getThemeNames(): string[];
+        newTheme(parentTheme: Theme, newTheme: ThemeConfig): Theme;
+        newThemeSet(): ThemeSet;
+        setDefault(opts: ThemeOpts, themeName: string): void;
+        setDefault(themeName: string): void;
+    }
+
+    class ThemeSetClass {
+        addTheme(themeName: string, parentThemeOrThemeObj: Theme, newTheme?: Theme | ThemeConfig): void;
+        addToAllThemes(theme: Theme): void;
+        getDefault(opts: ThemeOpts): Theme;
+        getTheme(name: string): Theme;
+        getThemeNames(): string[];
+        newTheme(parentTheme: Theme, newTheme: ThemeConfig): Theme;
+        newThemeset(): ThemeSet;
+        setDefault(opts: ThemeOpts, themeName: string): void;
+        setDefault(themeName: string): void;
+    }
+
+    interface ThemeConfig {
+        [key: string]: any;
+        activityIndicatorTheme?: string | string[];
+        postProgressbar?: string;
+        preProgressbar?: string;
+        preSubsection?: string;
+        progressBarTheme?: {
+            complete?: string;
+            remaining?: string;
+        };
+    }
+
+    interface Template {
+        align?: "left" | "right" | "center";
+        default?: string;
+        kerning?: number;
+        length?: number;
+        maxLength?: number;
+        minLength?: number;
+        type: string;
+        value?: string;
+    }
+
+    class Plumbing {
+        constructor(theme: Theme, template: Template, width: number);
+        setTheme(theme: Theme): void;
+        hide(): string;
+        hideCursor(): string;
+        setTemplate(template: Template): void;
+        setWidth(width: number): void;
+        show(status: string): string;
+        showCursor(): string;
+    }
+
+    interface ThemeOpts {
+        hasUnicode?: boolean;
+        hasColor?: boolean;
+        platform?: string;
+    }
+
+    interface StatusObject {
+        [key: string]: any;
+        completed: number;
+        section: string;
+        subsection: string;
+    }
+
+    interface GaugeConstructorOptions {
+        Plumbing?: Plumbing;
+        cleanupOnExit?: boolean;
+        enabled?: boolean;
+        fixedFramerate?: boolean;
+        hideCursor?: boolean;
+        template?: Template;
+        theme?: Theme | ThemeOpts | string;
+        themes?: ThemeSet;
+        tty?: WriteStream;
+        updateInterval?: number;
+    }
+
+    class GaugeClass {
+        constructor(streamOrOptions?: Stream | GaugeConstructorOptions, options?: GaugeConstructorOptions);
+        disable(): void;
+        enable(): void;
+        hide(callback?: () => void): void;
+        isEnabled(): boolean;
+        pulse(subsection?: string): void;
+        setTemplate(template: Template): void;
+        setTheme(theme: Theme | ThemeOpts | string): void;
+        setThemeset(themes: ThemeSet): void;
+        show(sectionOrStatus?: string | StatusObject, completed?: number): void;
+    }
+}
+
+declare module "gauge" {
```

Comment:
Please use a separate file for each of these, e.g. `gauge/index.d.ts`, `gauge/themes.d.ts`, `gauge/theme-set.d.ts`.